openai/openai.go

```diff
@@ -475,8 +477,24 @@ func fromChatRequest(r ChatCompletionRequest) (*api.ChatRequest, error) {
 		options["stop"] = stops
 	}
 
+	// Deprecated: MaxTokens is deprecated, use MaxCompletionTokens instead
 	if r.MaxTokens != nil {
-		options["num_predict"] = *r.MaxTokens
+		r.MaxCompletionTokens = r.MaxTokens
+	}
+
+	if r.NumCtx != nil {
+		options["num_ctx"] = *r.NumCtx
+	}
+
+	DEFAULT_NUM_CTX := 2048
+	// set num_ctx to max_completion_tokens if it's greater than num_ctx
+	if r.MaxCompletionTokens != nil {
+		options["num_predict"] = *r.MaxCompletionTokens
+		if r.NumCtx != nil && *r.MaxCompletionTokens > *r.NumCtx {
+			options["num_ctx"] = *r.MaxCompletionTokens
+		} else if *r.MaxCompletionTokens > DEFAULT_NUM_CTX {
+			options["num_ctx"] = *r.MaxCompletionTokens
```

Comment:
I think we can leave this as-is unless it's the expected behavior. Passing MaxCompletionTokens is fine as it can survive a context shift.